### ISSUE TITLE
about reg_loss

### ISSUE THREAD

**Issue by ycyinchao** [strongwolf/DW#26]
loc_loss = F.binary_cross_entropy(p_loc, torch.ones_like(p_loc), reduction='none') actually is equivalent to 5 * reg_loss
but,in the paper ,loc_loss should use giou_loss, but reg_loss has already used the giou_loss ，so What does 5 stand for？

**Comment by ycyinchao** [comment]
if reg_loss use iou_loss,it's right,because iou_loss defaultly use mode == 'log'（loss = -ious.log()），but in the paper and code, reg_loss use giou_loss,that means giou_loss defaultly use mode == 'linear'(loss = 1 - gious),through the formula，you cannot calculate iou, only giou

**Comment by ycyinchao** [comment]
[here](https://github.com/strongwolf/DW/blob/27f50e5bf8d9ea021d5fe1a7a7e47e06bc5db208/dw_head.py#L148)

**Comment by strongwolf** [comment]
5 means the reg_weight and it should be noticed that the reg_weight in the config file is always set as 1.
Like some other prediction-aware label assignment methods,  what we mean is that the positive weights should be related with the predition status of both classification and regression.  Actually, 'IoU' in the positive function of the paper should be understanded as the reg_score, which can be the predicted IoU (GFocal, VFocal) or exp(-reg_loss) (Autoassign, TOOD, FreeAnchor).  We have tried both and found that the latter performs slightly better.

**Comment by ycyinchao** [comment]
> if reg_loss use iou_loss,it's right,because iou_loss defaultly use mode == 'log'（loss = -ious.log()），but in the paper and code, reg_loss use giou_loss,that means giou_loss defaultly use mode == 'linear'(loss = 1 - gious),through the formula，you cannot calculate iou, only giou

![image](https://user-images.githubusercontent.com/105531816/193824132-e8b23e4c-f047-4fc2-bda2-e4621111f7d3.png)
through the method, use giou instead of iou maybe achieve better performance（more 0.2AP）
